zenoh/src/net/routing/hat/router/token.rs

```diff
@@ -386,8 +386,14 @@ fn send_forget_sourced_token_to_net_clildren(
 fn propagate_forget_simple_token(
     tables: &mut Tables,
     res: &Arc<Resource>,
+    src_face: Option<&Arc<FaceState>>,
     send_declare: &mut SendDeclare,
 ) {
+    dbg!(src_face.as_ref().map(|f| f.to_string()));
+    // FIXME(fuzzypixelz): This function should be split to accommodate for the divergent code paths
```

Comment:
Why should we do that ? Wouldn't this lead to a lot of code duplicate? What's wrong with an Option ?
(Note: if we do so, there are a lot of other functions that have an Optional src_face parameter that should also be changed)

Reply:
It was because of:

https://github.com/eclipse-zenoh/zenoh/blob/f7486160c01514d601186cb234f6edf49c76de63/zenoh/src/net/routing/hat/router/token.rs#L683

I thought that the face would need to stay `None` yet we don't want to send a token to the client/peer that intitiated the undeclaration.